File: easy/172/factorial_trailing_zeroes.py

class Solution:
    def trailingZeroes(self, n: int) -> int:
        r = (n - (n % 5)) // 5
        trailing_z = 0
        
        while r > 0:
            trailing_z += r
            r = r // 5
            
        return trailing_z

File: easy/172/factorial_trailing_zeroes.cpp

class Solution {
public:
    int trailingZeroes(int n) {
        int r = (n - (n % 5)) / 5;
        int trailing_z = 0;
        
        while (r > 0) {
            trailing_z += r;
            r = r / 5;
        }
        
        return trailing_z;
    }
};
